### PR TITLE
Update number of flakes notebook to template format

### PR DESCRIPTION
**Pull request by Shreyanand**
## Related Issues and Dependencies
#179 #161

## This introduces a breaking change

- [ ] Yes
- [X] No
- 
## Description
In this PR, I convert the number of flakes notebook to follow the metric template.
Note, I also update the template so that it's more efficient to use it as a base for writing all the notebooks.
In essence, I wrote a class for communicating to Ceph in metric_template.ipynb and changed number_of_flakes.ipynb to use functions and classes from the template.

### CODE REVIEW

**Review by aakankshaduggal — COMMENTED** [review]
/lgtm 
Really like how you wrote a class for communicating to Ceph. :+1:

File: notebooks/data-sources/TestGrid/metrics/number_of_flakes.ipynb [hunk 1]
```diff
@@ -18,31 +18,61 @@
   },
```
Comment (MichaelClifford) [inline notebooks/data-sources/TestGrid/metrics/number_of_flakes.ipynb:18]: <p>Can you use seaborn for both of the plots? and make them a bit larger</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aicoe-aiops/ocp-ci-analysis/pull/195/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (Shreyanand) [reply notebooks/data-sources/TestGrid/metrics/number_of_flakes.ipynb:18]: Changed :+1:

**Comment by Shreyanand** [comment]
* Added pyarrow to the Pipfile as it is required for conversion to the parquet format.

**Pull request merged**